### PR TITLE
Render code as pretty html instead of redirecting

### PR DESCRIPTION
**Pull request by mtharrison**
Here's a first pass at improving the experience of visiting scripts in the browser. This was part of the features wanted for 1.0 here: https://github.com/denoland/deno/issues/2473

Screenshot 2019-06-18 at 07 47 56

I had trouble setting up the stack for developing on this so I made a little mock server instead (https://gist.github.com/mtharrison/8ee409e56dc174d94b9d415d09c6a6b0) so this will need to be tested on lambda to make sure it works first.

### CODE REVIEW

**Comment by mtharrison** [comment]
This has an XSS risk, which I should address first.

File: src/app.js [hunk 1]
```diff
@@ -158,6 +159,27 @@ exports.lambdaHandler = async (event, context, callback) => {
   //const response = await fetchRemote(l);
   //callback(null, response);
 
+  if (
+    request.headers.accept &&
+    request.headers.accept.some(header => header.value.includes("text/html"))
+  ) {
+    const res = await fetch(l);
+    const body = await res.text();
```
Comment (ry) [inline src/app.js]: Can you use the `fetchRemote` method? it's commented out above.

File: src/app.js [hunk 2]
```diff
@@ -178,3 +200,59 @@ exports.lambdaHandler = async (event, context, callback) => {
   };
   callback(null, response);
 };
+
+function renderPrettyCode(pathname, code) {
+  const url = `https://deno.land${pathname}`;
```
Comment (kevinkassimo) [inline src/app.js]: Is this safe? `url` does not seem to be escapde in later contexts. Is pathname always properly escaped?

**Comment by ry** [comment]
can you use https://deno.land/style.css ?

**Comment by mtharrison** [comment]
Hey @ry sure, that looks better with that.
Screenshot 2019-06-18 at 07 47 56

**Comment by mtharrison** [comment]
> I wonder if you could make so that it doesn't redirect, even if it's not a real browser, but serves the code with "application/typescript" content type?

I could definitely look at that, should be straightforward, but just wondering what the benefit is? I guess it will increase your lambda costs if we're proxying all the code vs redirecting.

**Comment by mtharrison** [comment]
> I wonder if you could make so that it doesn't redirect, even if it's not a real browser, but serves the code with "application/typescript" content type?

I could definitely look at that, should be straightforward, but just wondering what the benefit is? I guess it will increase your lambda costs if we're proxying all the code vs redirecting.

**Comment by mtharrison** [comment]
> I wonder if you could make so that it doesn't redirect, even if it's not a real browser, but serves the code with "application/typescript" content type?

I could definitely look at that, should be straightforward, but just wondering what the benefit is? I guess it will increase your lambda costs if we're proxying all the code vs redirecting.

**Comment by ry** [comment]
> I could definitely look at that, should be straightforward, but just wondering what the benefit is? I guess it will increase your lambda costs if we're proxying all the code vs redirecting.

The cost is negligible. The benefit is
1) The HTML version and the text versions are more similar in functionality. 
2) When loading from a redirected URL, the deno progress bar displays the redirection target.. like this:
```
[66/66] Compiling https://raw.githubusercontent.com/denoland/deno_std/master/http/file_server.ts
```

I've tested this patch and it looks good! I'm landing as is. Thanks!

**Pull request merged**